src/classes/histogramSet.cpp

```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2025 Team Dissolve and contributors
+
+#include "classes/histogramSet.h"
+#include "base/lineParser.h"
+#include "classes/atomType.h"
+#include "items/deserialisers.h"
+#include "math/mathFunc.h"
+#include "templates/algorithms.h"
+
+HistogramSet::~HistogramSet()
+{
+    fullHistograms_.clear();
+    boundHistograms_.clear();
+    unboundHistograms_.clear();
+}
+
+/*
+ * Data
+ */
+
+// Initialise histograms
+void HistogramSet::initialise(const AtomTypeMix &atomTypeMix, double rdfRange, double binWidth)
+{
+    atomTypeMix_ = atomTypeMix;
+
+    auto nTypes = atomTypeMix_.nItems();
+
+    fullHistograms_.initialise(nTypes, nTypes, half_);
+    boundHistograms_.initialise(nTypes, nTypes, half_);
+    unboundHistograms_.initialise(nTypes, nTypes, half_);
+
+    dissolve::for_each_pair(
+        ParallelPolicies::par, nTypes,
+        [&](int i, int j)
+        {
+            fullHistograms_[{i, j}].initialise(0.0, rdfRange, binWidth);
+            boundHistograms_[{i, j}].initialise(0.0, rdfRange, binWidth);
+            unboundHistograms_[{i, j}].initialise(0.0, rdfRange, binWidth);
+        },
+        half_);
+}
+
+// Clear all histogram data
+void HistogramSet::clear()
+{
+    fullHistograms_.clear();
+    boundHistograms_.clear();
+    unboundHistograms_.clear();
+}
+
+// Zero histogram bins
+void HistogramSet::zeroBins()
+{
+    for (auto &histo : fullHistograms_.linearArray())
+        histo.zeroBins();
+    for (auto &histo : boundHistograms_.linearArray())
+        histo.zeroBins();
+    for (auto &histo : unboundHistograms_.linearArray())
+        histo.zeroBins();
```

Comment:
What do you think of a static method in `Array2D`, `flatten`, that takes a couple of `Array2D<T>`'s (like the above `full/bound/unboundHistograms_`) and flattens their `linearArray()`s into a single vector?
```suggestion
   for (auto &histo : Array2D::flatten({fullHistograms_, boundHistograms_, unboundHistograms_}))
        histo.zeroBins();
```

I believe the helper method could be implemented like:
```
static std::vector<A> flatten(const std::vector<Array2D<A>> arrays)
{
    std::vector<A> flat;
    for (const auto &a : arrays)
          flat.insert(flat.end(), a.begin(), a.end());
    return flat;
} 
```

Reply:
I see what you mean - in fact Adam's `zip()` operator would allow us to do this in one loop, albeit with three variables.  Your suggestion makes a copy of the data, so unfortunately wouldn't have the intended result. I think I will leave this as-is for now, but your suggestion makes me think that some kind of function which takes a lambda and operates on every element in the `Array2D` would be useful to implement in the future (e.g. `fullHistograms_.operate([](auto &histo) { histo.zeroBins(); })`)